lib/src/room.dart

```diff
@@ -2090,27 +2090,38 @@ class Room {
   /// Returns the [PushRuleState] for this room, based on the m.push_rules stored in
   /// the account_data.
   PushRuleState get pushRuleState {
-    final globalPushRules =
-        client.accountData['m.push_rules']?.content['global'];
-    if (globalPushRules is! Map) {
+    final globalPushRules = client.globalPushRules;
+    if (globalPushRules == null) {
       return PushRuleState.notify;
     }
 
-    if (globalPushRules['override'] is List) {
-      for (final pushRule in globalPushRules['override']) {
-        if (pushRule['rule_id'] == id) {
-          if (pushRule['actions'].indexOf('dont_notify') != -1) {
+    final overridePushRules = globalPushRules.override;
+    if (overridePushRules != null) {
+      for (final pushRule in overridePushRules) {
+        if (pushRule.ruleId == id) {
+          // "dont_notify" and "coalesce" should be ignored in actions since
+          // https://spec.matrix.org/v1.7/client-server-api/#actions
+          pushRule.actions
+            ..remove('dont_notify')
```

Comment:
You are removing this two times. Maybe as a cleaner approach just define a static variable and use it here and below instead of hardcoding two times. Just a personal prefrence.